stock_landed_costs_purchase_auto/models/stock_picking.py

```diff
@@ -0,0 +1,18 @@
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def action_done(self):
+        res = super().action_done()
+        for item in self.filtered(
+            lambda x: x.state == "done"
+            and x.picking_type_code == "incoming"
+            and x.purchase_id.landed_costs_ids
+        ):
+            for lc in item.purchase_id.landed_costs_ids:
+                lc.write({"picking_ids": [(6, 0, [item.id])]})
```

Comment:
This shouldn't be done here. Please link the picking on the picking creation.

Reply:
Ok, I have used the `_prepare_landed_cost_values()` function to set the linked `picking_ids`.